src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
+                resultBuilder,
             );
-        } catch (e) {
-            return falseIfGuardFailed(e);
+        } else {
+            await this.validateAndAddActs(input, resultBuilder, $.Action.Set);
         }
+        return;
     }
 
-    private isValueInRejectedValues(input: ControlInput): boolean {
+    private isSetWithoutValue(input: ControlInput): boolean {
         try {
-            okIf(InputUtil.isValueControlIntent(input, AmazonBuiltInSlotType.NUMBER));
-            const { action, target, values } = unpackValueControlIntent(
+            okIf(InputUtil.isIntent(input, GeneralControlIntent.name));
+            const { feedback, action, target } = unpackGeneralControlIntent(
                 (input.request as IntentRequest).intent,
             );
-            const valueStr = values[0].slotValue;
-            okIf(
-                InputUtil.actionIsMatchOrUndefined(action, [
-                    ...this.props.interactionModel.actions.set,
-                    ...this.props.interactionModel.actions.change,
-                ]),
-            );
             okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
-            okIf(this.state.rejectedValues.includes(Number.parseInt(valueStr, 10)));
-            this.handleFunc = this.handleValueExistsInRejectedValues;
+            okIf(InputUtil.feedbackIsMatchOrUndefined(feedback, FEEDBACK_TYPES));
+            okIf(InputUtil.actionIsMatch(action, this.props.interactionModel.actions.set));
+            this.handleFunc = this.handleSetWithoutValue;
             return true;
         } catch (e) {
             return falseIfGuardFailed(e);
         }
     }
 
-    private handleValueExistsInRejectedValues(
-        input: ControlInput,
-        resultBuilder: ControlResultBuilder,
-    ): void {
-        const { values } = unpackValueControlIntent((input.request as IntentRequest).intent);
-        const valueStr = values[0].slotValue;
-        this.setValue(valueStr);
-        resultBuilder.addAct(
-            new ProblematicInputValueAct(this, {
-                reasonCode: 'ValuePreviouslyRejected',
-                value: this.state.value,
-                renderedValue:
-                    this.state.value !== undefined ? this.props.valueRenderer(this.state.value, input) : '',
-            }),
-        );
-        resultBuilder.addAct(new RequestValueAct(this));
-    }
-
-    private isBareNoWhenConfirmingValue(input: ControlInput): boolean {
-        try {
-            okIf(InputUtil.isBareNo(input));
-            okIf(this.state.lastInitiative.actName === ConfirmValueAct.name);
-            this.handleFunc = this.handleFeedbackNoAndWithoutValueWhenConfirmingValue;
-            return true;
-        } catch (e) {
-            return falseIfGuardFailed(e);
-        }
+    private handleSetWithoutValue(input: ControlInput, resultBuilder: ControlResultBuilder): void {
+        this.askElicitationQuestion(input, resultBuilder, $.Action.Set);
```

Comment:
This one appears to meet the tenet of "necessary to understand the input".. i.e. it would be very weird to respond to "U: set my age" with something other than "A: to what value?".  So no need to refactor this to initiative phase.